packages/integration-tests/src/test-project-setup/seed_test_project.ts

```diff
@@ -0,0 +1,226 @@
+// it seems like as of amplify v6 , some of the code only runs in the browser ...
+// see https://github.com/aws-amplify/amplify-js/issues/12751
+if (process.versions.node) {
+  // node >= 20 now exposes crypto by default. This workaround is not needed: https://github.com/nodejs/node/pull/42083
+  if (new SemVer(process.versions.node).major < 20) {
+    // @ts-expect-error altering typing for global to make compiler happy is not worth the effort assuming this is temporary workaround
+    globalThis.crypto = crypto;
+  }
+}
+
+/**
+ * Creates test project for seed
+ */
+export class SeedTestProjectCreator implements TestProjectCreator {
+  readonly name = 'seed';
+
+  /**
+   * Creates project creator
+   */
+  constructor(
+    private readonly cfnClient: CloudFormationClient = new CloudFormationClient(
+      e2eToolingClientConfig,
+    ),
+    private readonly amplifyClient: AmplifyClient = new AmplifyClient(
+      e2eToolingClientConfig,
+    ),
+    private readonly cognitoIdentityProviderClient: CognitoIdentityProviderClient = new CognitoIdentityProviderClient(
+      e2eToolingClientConfig,
+    ),
+    private readonly stsClient: STSClient = new STSClient(
+      e2eToolingClientConfig,
+    ),
+  ) {}
+
+  createProject = async (e2eProjectDir: string): Promise<TestProjectBase> => {
+    const { projectName, projectRoot, projectAmplifyDir } =
+      await createEmptyAmplifyProject(this.name, e2eProjectDir);
+
+    const project = new SeedTestProject(
+      projectName,
+      projectRoot,
+      projectAmplifyDir,
+      this.cfnClient,
+      this.amplifyClient,
+      this.cognitoIdentityProviderClient,
+      this.stsClient,
+    );
+    await fsp.cp(
+      project.sourceProjectAmplifyDirURL,
+      project.projectAmplifyDirPath,
+      {
+        recursive: true,
+      },
+    );
+    return project;
+  };
+}
+
+class SeedTestProject extends TestProjectBase {
+  readonly sourceProjectDirPath = '../../src/test-projects/seed-test-project';
+
+  readonly sourceProjectAmplifyDirSuffix = `${this.sourceProjectDirPath}/amplify`;
+
+  readonly sourceProjectAmplifyDirURL: URL = new URL(
+    this.sourceProjectAmplifyDirSuffix,
+    import.meta.url,
+  );
+
+  constructor(
+    name: string,
+    projectDirPath: string,
+    projectAmplifyDirPath: string,
+    cfnClient: CloudFormationClient,
+    amplifyClient: AmplifyClient,
+    private readonly cognitoIdentityProviderClient: CognitoIdentityProviderClient,
+    private readonly stsClient: STSClient,
+  ) {
+    super(
+      name,
+      projectDirPath,
+      projectAmplifyDirPath,
+      cfnClient,
+      amplifyClient,
+    );
+  }
+
+  override async deploy(
+    backendIdentifier: BackendIdentifier,
+    environment?: Record<string, string>,
+  ) {
+    await super.deploy(backendIdentifier, environment);
+
+    const command = execaSync('npx', ['which', 'ampx'], {
+      cwd: this.projectDirPath,
+    }).stdout.trim();
+    const seedPolicyProcess = await execa(
```

Comment:
```suggestion
    const seedPolicyProcessResult = await execa(
```

if we await, process already exited.